interoperability-tests/Testing-between-Gramine-and-IntelSGXSDK.md

```diff
@@ -0,0 +1,147 @@
+```
+
+2. Build the sample
+
+```sh
+cd SampleCode/SampleAttestedTLS
+
+make
+```
+3. Start the server
+
+```sh
+./server/host/tls_server_host ./server/enc/tls_server_enclave.signed.so -port 4433
+
+```
+
+Now the tls_server_enclave listens on `127.0.0.1:4433`, with a self-signed X.509 certificate.
+
+Here is an example certificate generated by Intel SGX SDK RA-TLS server: [intel-sgxsdk-cert.pem](./intel-sgxsdk-cert.pem)
+
+## Gramine
+
+Gramine provides an example program [ra-tls-mbedtls](https://github.com/gramineproject/gramine/tree/master/CI-Examples/ra-tls-mbedtls) based on mbedTLS for RA-TLS. We use the client program in it for testing.
+
+1. Download and checkout correct Gramine version
+
+```sh
+git clone https://github.com/gramineproject/gramine.git
+cd gramine/
+git checkout dimakuv/interoperable-ra-tls
+```
+
+2. Build and install Gramine system-wide (in debug mode, so it can be GDBed if needed)
+
+```sh
+meson setup build-debug/ --buildtype=debug -Dskeleton=enabled -Ddirect=enabled -Dsgx=enabled -Ddcap=enabled -Dsgx_driver=upstream -Dsgx_driver_include_path=/usr/include/x86_64-linux-gnu/
+
+ninja -C build-debug/
+sudo ninja -C build-debug/ install
+```
+
+3. build and start the RA-TLS client
+
+```sh
+cd CI-Examples/ra-tls-mbedtls/
+make client
+RA_TLS_ALLOW_DEBUG_ENCLAVE_INSECURE=1 RA_TLS_ALLOW_OUTDATED_TCB_INSECURE=1  ./client dcap
+```
+
+> Use `RA_TLS_ALLOW_DEBUG_ENCLAVE_INSECURE=1` to allow debug enclaves, and use `RA_TLS_ALLOW_OUTDATED_TCB_INSECURE=1` to allow outdated TCB in quote.
+
+
+# Gramine as attester and Intel SGX SDK RA-TLS as verifier
+
+## Gramine
+
+We use the example program ra-tls-mbedtls provided by Gramine to start an RA-TLS server.
+
+1. Download and checkout correct Gramine version
+
+```sh
+git clone https://github.com/gramineproject/gramine.git
+cd gramine/
+git checkout dimakuv/interoperable-ra-tls
+```
+
+2. Build and install Gramine system-wide (in debug mode, so it can be GDBed if needed)
+
+```sh
+meson setup build-debug/ --buildtype=debug -Dskeleton=enabled -Ddirect=enabled -Dsgx=enabled -Ddcap=enabled -Dsgx_driver=upstream
+ninja -C build-debug/
+sudo ninja -C build-debug/ install
+```
+
+3. Build and start the RA-TLS server
+
+```sh
+cd CI-Examples/ra-tls-mbedtls/
+make app dcap RA_TYPE=dcap
+gramine-sgx ./server
+```
+
+Now the Gramine server listens on port 4433, clients can connect to it and examine its cert.
+
+Here is an example certificate generated by Gramine: [gramine-cert.pem](./gramine-cert.pem)
+
+## Intel SGX SDK RA-TLS
+
+There are 2 clients provided in Intel SGX SDK RA-TLS [SampleAttestedTLS](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS), client in trusted([client/](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS/client)) and client in untrusted([non_enc_client/](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS/non_enc_client)), both could be used to test RA-TLS connections. 
```

Comment:
`client in trusted` -> `client running inside the SGX enclave`

Reply:
updated.